src/Catel.Core/Catel.Core.Shared/Data/ModelBase.cs

```diff
@@ -312,6 +313,19 @@ protected internal bool LeanAndMeanModel
         public static bool DefaultDisableEventSubscriptionsOfChildValuesValue { get; set; }
 
         /// <summary>
+        /// Gets or sets a value indicating whether usage of equality comparer for equals should be disabled.
+        /// <para />
+        /// The default value is <c>false</c>.
+        /// </summary>
+        /// <value><c>true</c> if usage of equality comparer for equals should be disabled; otherwise, <c>false</c>.</value>
+        protected bool DisableUsageOfEqualityComparerForEquals { get; set; }
```

Comment:
Happy to go that path. We just need to make an important decision whether we will go 5.0 soon or have a 4.5 release in between. I am happy to put all these performance things into place, but that is definitely going to be 5.0 (since this will break existing stuff, but just for a minority of users). We are also working on Xamarin.Forms which is also planned for v5, and that is going very well, so we might just go v5 and work on it for 2-3 months.